ext/DifferentiationInterfaceEnzymeExt/reverse.jl

```diff
@@ -1,6 +1,15 @@
 const AutoReverseEnzyme = AutoEnzyme{Val{:reverse}}
 DI.autodiff_mode(::AutoReverseEnzyme) = DI.ReverseMode()
-DI.handles_output_type(::AutoReverseEnzyme, ::Type{<:AbstractArray}) = false
+
+struct Mutating{F}
+    f::F
+end
+
+function (m::Mutating{F})(y::AbstractArray, x) where {F}
```

Comment:
Doesn't Enzyme require `f!(x, y)` instead of `f!(y, x)`?
https://enzymead.github.io/Enzyme.jl/stable/generated/autodiff/#Defining-a-function

Or does it not matter?

Reply:
it doesn't matter